packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts

```diff
@@ -65,15 +72,28 @@ export function trackInteractionToNextPaint(
 
     const newInteraction = longestInteractions.estimateP98Interaction()
     if (newInteraction && newInteraction.duration !== interactionToNextPaint) {
+      const inpTarget = newInteraction.target
       interactionToNextPaint = newInteraction.duration
       interactionToNextPaintStartTime = elapsed(viewStart, newInteraction.startTime)
-
-      if (newInteraction.target && isElementNode(newInteraction.target)) {
-        interactionToNextPaintTargetSelector = getSelectorFromElement(
-          newInteraction.target,
-          configuration.actionNameAttribute
-        )
+      const isInpEnabled = isExperimentalFeatureEnabled(ExperimentalFeature.NULL_INP_TELEMETRY)
+
+      if (inpTarget && isElementNode(inpTarget)) {
+        interactionToNextPaintTargetSelector = getSelectorFromElement(inpTarget, configuration.actionNameAttribute)
+        if (!interactionToNextPaintTargetSelector && isInpEnabled) {
+          addTelemetryDebug('INP target selector is null', {
+            targetIsConnected: inpTarget.isConnected,
+            inp: newInteraction.duration,
+          })
+        }
       } else {
+        if (isInpEnabled) {
+          addTelemetryDebug('INP target is null or not an element node', {
+            hasTarget: !!inpTarget,
+            targetIsConnected: inpTarget ? inpTarget.isConnected : null,
+            inp: newInteraction.duration,
+          })
+        }
+
         interactionToNextPaintTargetSelector = undefined
```

Comment:
**💬 suggestion:** ‏to simplify a little bit:

```ts
      if (inpTarget && isElementNode(inpTarget)) {
        interactionToNextPaintTargetSelector = getSelectorFromElement(inpTarget, configuration.actionNameAttribute)
      } else {
        interactionToNextPaintTargetSelector = undefined
      }
      if (!interactionToNextPaintTargetSelector && isExperimentalFeatureEnabled(ExperimentalFeature.NULL_INP_TELEMETRY)) {
         addTelemetryDebug("Fail to get INP target selector", {
           hasTarget: !!inpTarget,
           targetIsConnected: inpTarget ? inpTarget.isConnected : undefined,
           targetIsElementNode: inpTarget ? isElementNode(inpTarget) : undefined,
           inp: newInteraction.duration
         })
      }
```

Also collect `isElementNode` so we cover the whole condition (`inpTarget && isElementNode(inpTarget)`)